x-pack/plugins/fleet/server/saved_objects/migrations/security_solution/to_v8_14_0.ts

```diff
@@ -35,3 +35,28 @@ export const migratePackagePolicyToV8140: SavedObjectModelDataBackfillFn<
 
   return { attributes: updatedPackagePolicyDoc.attributes };
 };
+
+export const migratePackagePolicyEnableCapsToV8140: SavedObjectModelDataBackfillFn<
+  PackagePolicy,
+  PackagePolicy
+> = (packagePolicyDoc) => {
+  if (packagePolicyDoc.attributes.package?.name !== 'endpoint') {
+    return { attributes: packagePolicyDoc.attributes };
+  }
+
+  const updatedPackagePolicyDoc: SavedObjectUnsanitizedDoc<PackagePolicy> = packagePolicyDoc;
+
+  const input = updatedPackagePolicyDoc.attributes.inputs[0];
+
+  if (input && input.config) {
+    const policy = input.config.policy.value;
+
+    const migratedPolicy = { enable_caps: true };
+
+    policy.linux.advanced = policy.linux.advanced
```

Comment:
I'd suggest to set the `enable_caps: true` field only if it doesn't exist yet. something like this could do it:
```ts
policy.linux.advanced = {
  ...policy.linux.advanced,
  events: {
    enable_caps: true,
    ...policy.linux.advanced?.events, // this comes second, so existing value precedes the added default
  },
};
```

this could help in cases, when in a managed environment a user sets the value to `false` on an updated Kibana instance, before every Kibana instance is updated and migration is performed. (@jloleysens could you please correct me if I'm wrong here?)

Reply:
d58e621b68afc760fc0e871d6a74faad03532d0e